src/main/java/org/folio/service/orders/OrderReEncumberService.java

```diff
@@ -258,8 +258,8 @@ private void adjustPoLineCost(CompositePoLine poLine, List<ReEncumbranceHolder>
       MonetaryAmount newFundAmount;
       if (fundDistr.getDistributionType().equals(FundDistribution.DistributionType.AMOUNT)) {
         MonetaryAmount fdAmount = Money.of(fundDistr.getValue(), poLine.getCost().getCurrency());
-        newFundAmount = fdAmount.divide(poLineEstimatedPriceBeforeRollover.getNumber().doubleValue())
-                                .multiply(poLineEstimatedPriceAfterRollover.getNumber().doubleValue());
+        newFundAmount = (poLineEstimatedPriceBeforeRollover.isZero()) ? fdAmount.divide(poLineEstimatedPriceBeforeRollover.getNumber().doubleValue())
```

Comment:
shouldn't it be !poLineEstimatedPriceBeforeRollover.isZero() ?

Reply:
definitely :)